src/AlignmentVault.sol

```diff
@@ -408,32 +414,44 @@ contract AlignmentVault is Ownable, Initializable, ERC721Holder, ERC1155Holder,
     ) external payable virtual onlyOwner returns (uint256 positionId) {
         positionId = _NFTX_INVENTORY.depositWithNFT(vaultId, tokenIds, amounts, address(this));
         _inventoryPositionIds.add(positionId);
-        emit AV_InventoryPositionCreated(positionId, _countTokens(tokenIds, amounts) * 1 ether);
+        (,,,,, uint256 vTokenShares,,) = _NFTX_INVENTORY.positions(positionId);
+        emit AV_InventoryPositionCreated(positionId, vTokenShares);
     }
 
     // Only works on inventory positions created with vTokens
     function inventoryPositionIncrease(uint256 positionId, uint256 vTokenAmount) external payable virtual onlyOwner {
+        (,,,,, uint256 vTokenSharesBefore,,) = _NFTX_INVENTORY.positions(positionId);
         _NFTX_INVENTORY.increasePosition(positionId, vTokenAmount, "", false, true);
-        emit AV_InventoryPositionIncreased(positionId, vTokenAmount);
+        (,,,,, uint256 vTokenSharesAfter,,) = _NFTX_INVENTORY.positions(positionId);
+        emit AV_InventoryPositionIncreased(positionId, vTokenSharesAfter - vTokenSharesBefore);
     }
 
-    // vTokenAmount must include `tokenIds.length * 1 ether` if any NFTs are to be withdrawn
+    // value of vTokenShares must >= `tokenIds.length * 1 ether` if any NFTs are to be withdrawn
     function inventoryPositionWithdrawal(
         uint256 positionId,
-        uint256 vTokenAmount,
+        uint256 vTokenShares,
         uint256[] calldata tokenIds,
         uint256 vTokenPremiumLimit
     ) external payable virtual onlyOwner {
+        if (
+            tokenIds.length != 0 &&
+            (tokenIds.length * 1 ether) > _NFTX_INVENTORY.pricePerShareVToken(vaultId)
+        ) revert AV_InsufficientSharesToWithdrawNft();
         if (vTokenPremiumLimit == 0) vTokenPremiumLimit = type(uint256).max;
-        _NFTX_INVENTORY.withdraw(positionId, vTokenAmount, tokenIds, vTokenPremiumLimit);
-        emit AV_InventoryPositionWithdrawal(positionId, vTokenAmount);
+        _NFTX_INVENTORY.withdraw(positionId, vTokenShares, tokenIds, vTokenPremiumLimit);
+        // review: on full withdraw should we remove position from set (doesn't seem like position is burnable)?
```

Comment:
No, the position should not be removed from the set if the position is not burned, as it can always be added to later.